### DISCUSSION TITLE
How exactly does assigning colors work on the arturia minilab mkII?

### DISCUSSION THREAD

**Discussion by on3iro** [helgoboss/helgobox#1110] [Q&A] [answered]
Hey,

I am currently trying to figure out, how assigning colors to the Minilab Pads works exactly.
I have two scenarios - one almost works, the other I have no idea how to achieve it:

1. I want to have colored transport controls. This does almost work as intended, however it would be really nice if there where some kind of feedback when they are toggled on. Here's a screenshot of my `Play/Stop` button for example:

![image](https://github.com/user-attachments/assets/23b4408d-a39d-4ccc-b0ae-984a60bc85d0)

2. I want to have simpel colore buttons to switch between tracks. This doesn't work at all, yet and I am not quite sure what I would have to specify as target, to make it work. Here's my `Previous Track` button as an example:

![image](https://github.com/user-attachments/assets/5ce44c8a-4589-48f7-9f10-f9f9c58d28e1)

Any help is appreciated :)

**Comment by helgoboss** [comment, marked as answer]
> Hey,
> 
> I am currently trying to figure out, how assigning colors to the Minilab Pads works exactly. I have two scenarios - one almost works, the other I have no idea how to achieve it:
> 
> 1. I want to have colored transport controls. This does almost work as intended, however it would be really nice if there where some kind of feedback when they are toggled on. Here's a screenshot of my `Play/Stop` button for example:

Read this: https://github.com/helgoboss/helgobox/wiki/ReaLearn-Controllers#arturia-minilab-mkii. You need to use the MiniLab controller preset and then change source min/max values to get the colors you want ... the min value color is used for "off" and the max value color for "on".

> 2. I want to have simpel colore buttons to switch between tracks. This doesn't work at all, yet and I am not quite sure what I would have to specify as target, to make it work. Here's my `Previous Track` button as an example:

Let's first talk about the control direction before talking about color feedback. You need target "Project: Browse tracks" and use mode "Incremental button". For the "Previous button", you also need to check "Reverse".

**Reply by on3iro** [reply]
Oh, I see - thanks. I thought I could simply use the actions but of course these won't receive the proper feedback. Thanks!
